src/configure/utilities/InputControlProvider.ts

```diff
@@ -0,0 +1,180 @@
+import { MustacheHelper } from "../helper/mustacheHelper";
+import { ExtendedInputDescriptor, ExtendedPipelineTemplate, InputDataType, InputMode } from "../model/Contracts";
+import { AzureSession, ControlType, IPredicate, StringMap } from '../model/models';
+import { DataSourceExpression } from "./DataSourceExpression";
+import { InputUxDescriptor } from "./InputUxDescriptor";
+import { InputUxDescriptorUtility } from "./InputUxDescriptorUtility";
+import { VisibilityHelper } from "./VisibilityHelper";
+
+export class InputControlProvider {
```

Comment:
Is InputControlProvider really an utility class? I feel it provides real functionality in e2e flow. What do you think? Similarly for other related classes you have added in `utilities` folder. If you are keeping them in this folder only, then create a new sub-folder for them.

Reply:
I think that this is a utility to get the inputs for template parameters. Moved it to sub-folder templateInputsHelper

Reply:
In my opinion, a utility is a static kind of class which provides a context-less logic to manipulate some data. for example, getting resource name from reosurce id is a utility.
If we make controlProvider as an utility then by that logic everything can become an utility. for example, configurer can become an utility - an utility which configures a pipeline. How do we differentiate?

Reply:
I got your point, created templateInputHelper folder to have all the files related to input provider and it also contains the required utilities to parse input descriptor and get the inputs.